internal/rbd/group/volume_group.go

```diff
@@ -35,16 +35,16 @@ import (
 
 var ErrRBDGroupNotConnected = errors.New("RBD group is not connected")
 
-// volumeGroup handles all requests for 'rbd group' operations.
-type volumeGroup struct {
-	// id is a unique value for this volume group in the Ceph cluster, it
+// VolumeGroup handles all requests for 'rbd group' operations.
+type VolumeGroup struct {
```

Comment:
Don't export these, they are private intentional. They implement the interface in the `internal/rbd/types/*.go` files. You can use `rbd.NewManager()` to get a manager struct and use the Manager API to get the VolumeGroup.

Reply:
If you are referring to `GetVolumeGroupByID`, then It doesn't suite well to use it here. Because in `GetVolumeGroup` we have GetPoolName by `LocationID` (which is PoolID) and PoolID may be different in secondary cluster. 

https://github.com/ceph/ceph-csi/blob/130e8b42f2ef72bd54e8dc0f96823af3c3890ee8/internal/rbd/group/volume_group.go#L106-L109

Else we can do it by passing an optional parameter `PoolName` for `GetVolumeGroup` ?
If it is passed as empty, then call `GetPoolName` else use the `PoolName` passed?